ui/src/contexts/Identity.tsx

```diff
@@ -0,0 +1,157 @@
+import { message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import { getIdentities } from "src/adapters/api/identities";
+import { useEnvContext } from "src/contexts/Env";
+import { AppError, Identity } from "src/domain";
+import { ROUTES } from "src/routes";
+import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { IDENTIFIER_SEARCH_PARAM, ROOT_PATH } from "src/utils/constants";
+import { buildAppError } from "src/utils/error";
+
+type IdentityState = {
+  fetchIdentities: (signal: AbortSignal) => void;
+  handleChange: (identifier: string) => void;
```

Comment:
This is selecting an identity, right? Can we better name it? Maybe `selectIdentity`?

Reply:
but we keep there a list and few function, I am not sure that fetchIdentities connected to selectIdentity name

Reply:
Agreed on a meet.